### ISSUE TITLE
[BUG] Kotlin 1.6.20 causes JobMethodNotFoundException

### ISSUE THREAD

**Issue by theexiile1305** [jobrunr/jobrunr#417]
**Are you running the latest version of JobRunr?**
Yes!

**Describe the bug**
Upgrading to Kotlin version `1.6.20` with Jobrunr version `5.0.0` produces the error  `JobMethodNotFoundException` with the following stack trace:
```
2022-04-12 15:51:10.760 ERROR 82323 --- [pool-3-thread-1] o.jobrunr.server.BackgroundJobPerformer  : Job(id=76f532b2-5b1a-4a44-9712-1fc932e489e8, jobName='null') processing failed: Job method not found

org.jobrunr.scheduling.exceptions.JobMethodNotFoundException: com.example.service..process$lambda-0(java.lang.String, java.lang.String)
	at org.jobrunr.utils.JobUtils.lambda$getJobMethod$0(JobUtils.java:45) ~[jobrunr-5.0.0.jar:5.0.0]
	at java.base/java.util.Optional.orElseThrow(Optional.java:403) ~[na:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:45) ~[jobrunr-5.0.0.jar:5.0.0]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:40) ~[jobrunr-5.0.0.jar:5.0.0]
	at org.jobrunr.utils.JobUtils.assertJobExists(JobUtils.java:49) ~[jobrunr-5.0.0.jar:5.0.0]
	at org.jobrunr.server.BackgroundJobServer.getBackgroundJobRunner(BackgroundJobServer.java:217) ~[jobrunr-5.0.0.jar:5.0.0]
	at org.jobrunr.server.BackgroundJobPerformer.runActualJob(BackgroundJobPerformer.java:83) ~[jobrunr-5.0.0.jar:5.0.0]
	at org.jobrunr.server.BackgroundJobPerformer.run(BackgroundJobPerformer.java:43) ~[jobrunr-5.0.0.jar:5.0.0]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
```

**Environment**
I'm using JobRunr version: `5.0.0`
I'm running on JDK: `asdf`
I'm using Kotlin version `1.6.20`.
I'm using the following StorageProvider: PostgreSQL 

**How to Reproduce**
Just using the latest Jobrunr package `5.0.0` with kotlin version `1.6.20`

**Expected behavior**
Jobrunr should be without problems with kotlin version `1.6.20`.

**Comment by pixnbit** [comment]
I just found this library and I was reading through the documentation. Persisting java lambda is a disaster in the making for errors like this, by JVM updates or your own code base refactor/rename. I think the limitations should be stated in the documentation, serious production code should not use lambda. JobRequests is a much safer way albeit more verbose.

**Comment by rdehuyss** [comment]
@pixnbit - well, I agree with you regarding Kotlin. They change the underlying Java generated bytecode quite often. For Java however, JobRunr takes part in the Oracle JDK Quality Outreach Program and this makes sure that it keeps working in newer versions of the JVM.

**Comment by laguiar** [comment]
@rdehuyss maybe something like a kotlin extension might be a good improvement idea?
also Kotlin examples in the documentation.
using `JobRequest` is not much Kotlin _friendly_.

**Comment by rdehuyss** [comment]
I'm looking into this with Jetbrains. See https://youtrack.jetbrains.com/issue/KT-52008

**Comment by theexiile1305** [comment]
With the new kotlin release of version `1.6.21 `this error also occurs.

**Comment by rdehuyss** [comment]
Fixed with 8cde1ddb6ccfc11d342fc63685a6f0f203cc682d
